### DISCUSSION TITLE
Deepstream plugin failed

### DISCUSSION THREAD

**Discussion by rgbsi-dk1** [OE4T/meta-tegra#911] [Q&A] [answered]
Hi guys,
I'm not able to to build core-image-sato-dev image with deepstream python plugin for jetson-nano-emmc machine using dunfell-l4t-32.5.0 branch.
I was going through this link [https://developer.ridgerun.com/wiki/index.php?title=Yocto_Support_for_NVIDIA_Jetson_Platforms_-_DeepStream](url). As Its been mentioned, deepstream python binding depends on python3.6 So, I added **PREFERRED_VERSION_python3 = "3.6%"**, **PREFERRED_VERSION_python3-native = "3.6%"**  and **require contrib/conf/include/gstreamer-1.14.conf** in local.conf but i got a lots of warnings and error while build. 

 bitbake core-image-sato-dev
Loading cache: 100% |########################################################################################################################################################################| Time: 0:00:01
Loaded 3726 entries from dependency cache.
Parsing recipes: 100% |######################################################################################################################################################################| Time: 0:00:02
Parsing of 2484 .bb files complete (2356 cached, 128 parsed). 3720 targets, 159 skipped, 0 masked, 0 errors.
Removing 1 recipes from the aarch64 sysroot: 100% |##########################################################################################################################################| Time: 0:00:00
NOTE: Resolving any missing task queue dependencies
WARNING: preferred version 3.6% of python3-native not available (for item python3-native)
WARNING: versions of python3-native available: 3.8.12
WARNING: preferred version 3.6% of python3-native not available (for item python3-core-native)
WARNING: versions of python3-native available: 3.8.12
...
...

Build Configuration:
BB_VERSION           = "1.46.0"
BUILD_SYS            = "x86_64-linux"
NATIVELSBSTRING      = "universal"
TARGET_SYS           = "aarch64-poky-linux"
MACHINE              = "jetson-nano-devkit-emmc"
DISTRO               = "poky"
DISTRO_VERSION       = "3.1.14"
TUNE_FEATURES        = "aarch64 armv8a crc"
TARGET_FPU           = ""
meta-tegra           
contrib              = "dunfell-l4t-r32.5.0:b89edb23da0802306e8be2f8f2b267a0128305ae"
meta                 
meta-poky            
meta-yocto-bsp       = "dunfell:bba323389749ec3e306509f8fb12649f031be152"
meta-rgbsi           = "<unknown>:<unknown>"
meta-oe              
meta-perl            
meta-python          = "dunfell:ab9fca485e13f6f2f9761e1d2810f87c2e4f060a"
meta-virtualization  = "dunfell:c5f61e547b90aa8058cf816f00902afed9c96f72"
meta-filesystems     
meta-networking      = "dunfell:ab9fca485e13f6f2f9761e1d2810f87c2e4f060a"
workspace            = "<unknown>:<unknown>"

Initialising tasks: 100% |###################################################################################################################################################################| Time: 0:00:05
Sstate summary: Wanted 70 Found 3 Missed 67 Current 3520 (4% match, 98% complete)
NOTE: Executing Tasks
ERROR: gstreamer1.0-python-1.14.4-r0 do_configure: configure failed

...
...

checking pkg-config is at least version 0.9.0... yes
| checking for gstreamer-1.0 >= 1.14.4... yes
| checking for pygobject-3.0 >= 3.8... yes
| okay
| checking for headers required to compile python extensions... found
| checking for pygobject overrides directory... ${libdir}/python3.8/site-packages/gi/overrides
| checking for gstreamer-1.0 >= 1.14.4... yes
| configure: Using /usr/lib/gstreamer-1.0 as the plugin install location
| checking for pygobject-3.0 >= 3.8... yes
| checking for libraries required to embed python... no
| configure: error: Python libs not found. Windows requires Python modules to be explicitly linked to libpython.
| NOTE: The following config.log files may provide further information.
| NOTE: /home/rgbsi/Workspace/yocto/build_nano_sato_dev/tmp/work/aarch64-poky-linux/gstreamer1.0-python/1.14.4-r0/build/config.log
| ERROR: configure failed
| WARNING: exit code 1 from a shell command.

Its seems there is no recipe for python3.6 so its taking default python3.8 recipe but due to this error "configure: error: Python libs not found. Windows requires Python modules to be explicitly linked to libpython." build getting failed

I just want to add deepstream python support in my image but struct in this problem. Please help me to proceed.

**Comment by madisongh** [comment, marked as answer]
Unfortunately, you cannot use the DeepStream Python bindings natively in a Yocto-built image due to their dependency on Python 3.6. If you really need Python+DeepStream, you could run them in a container. You could also contact RidgeRun, as mentioned on their wiki page, to ask about other alternatives.

NVIDIA released source code for the Python bindings for DeepStream 6.0, so in theory, at least, we could have a recipe to build them, and that should make them compatible with whatever version of Python we have.  No one has submitted a recipe for them yet.

There does appear to be an issue with the gstreamer1.0-python 1.14 recipe we have in the contrib layer, which I'll take a look at.

**Reply by madisongh** [reply]
FYI, I spent some time on this, and we now have a recipe for the [DeepStream 6.0 Python bindings](https://github.com/OE4T/meta-tegra-community/tree/dunfell/recipes-devtools/deepstream/deepstream-6.0-pyds) in meta-tegra-community (dunfell and master branches).  I've also resolved the issue with the `gstreamer1.0-python`  plugin recipe we have in meta-tegra/contrib.
